### PR TITLE
add seer

### PR DESCRIPTION
**Pull request by xyzseer**
##### Name (to be shown on DefiLlama):
Seer

##### Twitter Link:
https://x.com/seer_pm

##### List of audit links if any:
https://app.hats.finance/audit-competitions/seer-pm-0x899bc13919880db76edf4ccd72bdfa5dfa666fb7/rewards

##### Website Link:
https://seer.pm/

##### Logo (High resolution, will be shown with rounded borders):
Seer-logo

##### Current TVL:
684k

##### Chain:
Gnosis / Ethereum

##### Short Description (to be shown on DefiLlama):
A decentralized protocol allowing users to create markets, trade yield bearing tokens, use outcome tokens compatible with other exchanges, and offering new types of prediction markets.

##### Category (full list at https://defillama.com/categories) *Please choose only one:
Prediction Market

##### methodology (what is being counted as tvl, how is tvl being calculated):
TVL represents the total quantity of sDAI held in the conditional tokens contract. The sDAI is withdrawn when the participants merge or redeem their tokens.

##### Github org/user (Optional, if your code is open source, we can track activity):
https://github.com/seer-pm

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/seer exports TVL: 
        
 
 
    xdai                      482.68 k
    ethereum                  202.05 k
    
    total                    684.72 k

**Comment by g1nt0ki** [comment]
@xyzseer thanks for the PR, is there some other way to pull the data on chain other than relying on logs? as the #logs will keep going up, so the adapter will run into performance issues in a few months.

**Comment by llamatester** [comment]
Error while running adapter at projects/seer: 
        
 
  
    
    
    Error: Promise pool failed! 
     PromisePoolError: invalid address (argument="address", value=null, code=INVALID_ARGUMENT, version=6.13.1) (argument="marketFactory", value=null, code=INVALID_ARGUMENT, version=6.13.1)

**Comment by llamatester** [comment]
The adapter at projects/seer exports TVL: 
        
 
 
    xdai                      502.26 k
    ethereum                  200.13 k
    
    total                    702.39 k

**Comment by xyzseer** [comment]
@g1nt0ki fixed, let me know if it's ok :)

File: projects/seer/index.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+ */
+function calculateTotalSupply(marketsData, outcomeTokensSupplyByMarket) {
+  const marketSupplies = new Map();
+  const processedTokens = new Set();
+
+  marketsData.forEach((market, index) => {
+    const supply = outcomeTokensSupplyByMarket[index];
+    const uniqueSupply = supply.filter((_, i) => !processedTokens.has(market.wrappedTokens[i]));
+
+    uniqueSupply.forEach((_, i) => processedTokens.add(market.wrappedTokens[i]));
```
Comment (g1nt0ki) [inline projects/seer/index.js:87]: I think this is a bug? it should be `supply` else we add the wrong token?

**Comment by llamatester** [comment]
The adapter at projects/seer exports TVL: 
        
 
 
    xdai                      503.19 k
    ethereum                  200.21 k
    
    total                    703.40 k

**Comment by g1nt0ki** [comment]
@xyzseer can you check my refactor? main change is batching the calls

**Comment by xyzseer** [comment]
thanks! it looks good

**Pull request merged**